### PR TITLE
Pruning ratio + pruned weights distribution plotting

### PR DESCRIPTION
**Pull request by mbronckers**
Added dimensionality members in BaseBFC to help with masking sample draws with prune mask.

### CODE REVIEW

File: src/bbb/models/bnn.py [hunk 1]
```diff
@@ -374,6 +374,27 @@ def prune_weights(self, weight_fill_masks: List[Tensor]):
...
+        non_pruned_samples = []
+    
+        weight_samples = self.weight_samples()
+        for layer_samples, layer, layer_mask in zip(weight_samples, bfc_layers, weight_fill_masks):
+        
+            ls = torch.reshape(layer_samples, shape=(self.inference_samples, layer.dim_in, layer.dim_out))
+            non_pruned_samples_layer = []
+        
+            for i, sample in enumerate(ls):
+                non_pruned_weights = sample[~layer_mask.T]
```
Comment (thatscotdatasci) [inline src/bbb/models/bnn.py:389]: How come we need to transpose the mask? Seems strange...

Comment (mbronckers) [reply src/bbb/models/bnn.py:389]: The order of the mask is wrong, the dimensions were flipped (mask was 1200 x 786) whereas dim_in x dim_out = (786x1200)

Comment (thatscotdatasci) [reply src/bbb/models/bnn.py:389]: Would be good to understand _how_ the order of the mask ended up being wrong; I don't reshape at any point as far as I remember.

Comment (thatscotdatasci) [reply src/bbb/models/bnn.py:389]: Yeah, the weights are (dim_out, dim_in)

image

**Comment by thatscotdatasci** [comment]
Couldn't seem to see your updates in the notebook. However, this was working for me:

```python
snr_threshold_quantiles = np.arange(0,1,0.1)
snr_threshold_accuracies = np.zeros_like(snr_threshold_quantiles)

for i, snr in enumerate(snr_threshold_quantiles):
    snr_threshold = np.quantile(weight_snrs, snr)
    weight_fill_masks = [
        weight_snr_tensors[i] < snr_threshold for i in range(len(weight_snr_tensors))
    ]
    net.prune_weights(weight_fill_masks)
    eval_dict = evaluate_classifier(net, X_val)
    print(eval_dict['eval_score'].item())
    snr_threshold_accuracies[i] = eval_dict['eval_score'].item()
```

**Pull request merged**